src/main/java/com/everis/cars/exceptions/BeanValConstrainViolationExceptionMapper.java

```diff
@@ -1,27 +1,56 @@
 package com.everis.cars.exceptions;
 
+import javax.interceptor.Interceptors;
 import javax.validation.ConstraintViolation;
 import javax.validation.ConstraintViolationException;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
 import javax.ws.rs.ext.Provider;
 
+import org.apache.log4j.Logger;
+
 import com.everis.cars.entity.ErrorMessage;
 import com.everis.cars.entity.ErrorMessageCollection;
+import com.everis.cars.interceptors.LoggerInterceptor;
 
+/**
+ * ExceptionMapper to catch Bean Validation exceptions and standardize exception responses
+ */
 @Provider
+@Interceptors(LoggerInterceptor.class)
 public class BeanValConstrainViolationExceptionMapper implements ExceptionMapper<ConstraintViolationException>  {
 
+	/**
+	 * Logger instance
+	 * 
+	 * @see org.apache.log4j.Logger
+	 */
+	private static Logger logger = Logger.getLogger(BeanValConstrainViolationExceptionMapper.class);
+	
+	/**
+	 * toResponse override to implement the custom response message 
```

Comment:
what the main purpose of this?

Reply:
rest layer to throw 400 when...

Reply:
I explained better this methods in exception mappers.